src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -2448,21 +2606,35 @@ def _update_addons(cmd,  # pylint: disable=too-many-branches,too-many-statements
 
     addon_profiles = instance.addon_profiles or {}
     if 'kube-dashboard' in addon_args and 'kubeDashboard' not in addon_profiles:
-        addon_profiles['kubeDashboard'] = ManagedClusterAddonProfile(enabled=True)
+        addon_profiles['kubeDashboard'] = ManagedClusterAddonProfile(
+            enabled=True)
 
     os_type = 'Linux'
 
     # for each addons argument
     for addon_arg in addon_args:
-        addon = ADDONS[addon_arg]
+        if addon_arg in ADDONS:
+            addon = ADDONS[addon_arg]
+        else:
+            matches = fuzzy_match(addon_arg, list(ADDONS))
+            matches = str(matches)[1:-1]
+            if len(matches) > 0:
+                raise KeyError(
```

Comment:
Can we raise `CLIError` instead of `KeyError` here ?

Reply:
Yes sir!